domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/ValidUnemploymentConstraintValidator.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.cmc.domain.constraints;
+
+import uk.gov.hmcts.cmc.domain.models.statementofmeans.Unemployment;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.mayNotBeProvidedError;
+import static uk.gov.hmcts.cmc.domain.constraints.utils.ConstraintsUtils.setValidationErrors;
+
+public class ValidUnemploymentConstraintValidator implements ConstraintValidator<ValidEmployment, Unemployment> {
+
+    static class Fields {
+        static final String UNEMPLOYED = "unemployed";
+        static final String IS_RETIRED = "isRetired";
+        static final String OTHER = "other";
+    }
+
+    @Override
+    public boolean isValid(Unemployment unemployment, ConstraintValidatorContext context) {
+        boolean valid = true;
+        boolean isRetiredPopulated = unemployment.isRetired();
+        boolean otherPopulated = unemployment.getOther().isPresent();
+        boolean unemployedPopulated = unemployment.getUnemployed().isPresent();
+
+        if (isRetiredPopulated) {
+            if (otherPopulated) {
+                setValidationErrors(context, Fields.OTHER, mayNotBeProvidedError(Fields.OTHER, Fields.IS_RETIRED));
```

Comment:
This should be `mayNotBeProvidedError(Fields.OTHER` replaced with `mayNotBeProvidedError('unemployment'`.

We want to say here that other is not allowed when unemployment is set to retired.

Reply:
done